lib/bolt/project.rb

```diff
@@ -44,13 +45,23 @@ def self.find_boltdir(dir)
 
     def self.create_project(path, type = 'option')
       fullpath = Pathname.new(path).expand_path
+
+      if ENV['BOLT_PROJECT'].nil? && fullpath.world_writable?
+        raise Bolt::Error.new(
+          "Project directory '#{fullpath}' is world-writable. Bolt will not load projects from "\
```

Comment:
I think this needs to clarify a couple of a) that this isn't used for security reasons, b) exactly how to set `BOLT_PROJECT` to do what you want. Also "Bolt will not load projects from [...]" sounds like Bolt will _run_ but won't load the project, which isn't accurate. Maybe something like this:

```
"Project directory '#{fullpath}' is world-writable which poses a security risk. Set BOLT_PROJECT='${fullpath}' to force the use of this project directory."
```